core/shared/src/main/scala/canoe/api/sources/Polling.scala

```diff
@@ -29,10 +29,7 @@ private[api] class Polling[F[_]: TelegramClient: Functor](timeout: FiniteDuratio
       .map(updates => (lastId(updates).map(_ + 1).getOrElse(offset), updates))
 
   private def lastId(updates: List[Update]): Option[Long] =
-    updates match {
-      case Nil      => None
-      case nonEmpty => Some(nonEmpty.map(_.updateId).max)
-    }
+    updates.map(_.updateId).maxOption
```

Comment:
```suggestion
    updates.lastOption.map(_.updateId)
```
is even better as it won't allocate a new list

Reply:
If updates comes already sorted by id, then I am fully agree with you!

Reply:
@ivan-klass it seems that the owner is unavailable for questions. I am struggling to understand some parts of the canoe's code, can I ask you about such? I didn't find any contact data in your profile, so how can I reach you?